File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.c
int removeDuplicates(int* nums, int numsSize) {
    // if(numsSize <= 2){
    //     return numsSize;
    // }

    // int prev = 1;
    // int curr = 2;
    // while(curr < numsSize){
    //     if(nums[curr] == nums[prev] && nums[curr] == nums[prev - 1]){
    //         curr++;
    //     }else{
    //         prev++;
    //         nums[prev] = nums[curr];
    //         curr++;
    //     }
    // }
    // return prev + 1;

    if(numsSize <= 2){
        return numsSize;
    }

    int k = 2;
    for(int i = 2; i < numsSize; i++){
        if(nums[i] != nums[k - 2]){
            nums[k++] = nums[i];
        }
    }
    return k;
}

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        if len(nums) <= 2:
            return len(nums)
        curr = 2
        prev = 1
        while curr < len(nums):
            if nums[curr] == nums[prev] and nums[curr] == nums[prev - 1]:
                curr += 1
            else:
                prev += 1
                nums[prev] = nums[curr]
                curr += 1
        return prev + 1